### ISSUE TITLE
[dc.js migration] auto-selected `graph.x_axis.axis_enabled` setting does not match old implementation

### ISSUE THREAD

**Issue by EmmadUsmani** [metabase/metabase#38783]
Similar to https://github.com/metabase/metabase/issues/38782, this was also something that came up often in [#echarts-bug-bash](https://metaboat.slack.com/archives/C06G94JTWBS), below are some examples.

https://metaboat.slack.com/archives/C06G94JTWBS/p1706193475366799

https://metaboat.slack.com/archives/C06G94JTWBS/p1706193183854899

https://metaboat.slack.com/archives/C06G94JTWBS/p1706193082481509

Below is the logic that the old implementation used

```javascript
function beforeRenderComputeXAxisLabelType(chart) {
  // treat graph.x_axis.axis_enabled === true as "auto"
  if (chart.settings["graph.x_axis.axis_enabled"] === true) {
    const overlaps = checkXAxisLabelOverlap(chart);
    if (overlaps) {
      if (chart.isOrdinal()) {
        const spacing = computeXAxisSpacing(chart);
        if (spacing < X_LABEL_HIDE_THRESHOLD) {
          chart.settings["graph.x_axis.axis_enabled"] = false;
        } else if (spacing < X_LABEL_ROTATE_90_THRESHOLD) {
          if (checkLabelHeight(chart, 90)) {
            chart.settings["graph.x_axis.axis_enabled"] = "rotate-90";
          } else {
            chart.settings["graph.x_axis.axis_enabled"] = false;
          }
        } else {
          if (checkLabelHeight(chart, 45)) {
            chart.settings["graph.x_axis.axis_enabled"] = "rotate-45";
          } else {
            chart.settings["graph.x_axis.axis_enabled"] = false;
          }
        }
      } else {
        chart.settings["graph.x_axis.axis_enabled"] = false;
      }
    }
  }
}
```

To summarize, for numeric (linear, log, power) and timeseries x-axes, if labels will overlap then the axis will be hidden. For categorical (ordinal, historgarm) x-axes, if labels overlap it will try to use 90 degree or 45 degree rotated labels, or it will hide them if those overlap as well.

### Examples

question/7769
https://metaboat.slack.com/archives/C06G94JTWBS/p1706194101957459

question/14061
https://metaboat.slack.com/archives/C06G94JTWBS/p1706194432975859

question/97
https://metaboat.slack.com/archives/C06G94JTWBS/p1706194969563009
